### PR TITLE
✨ Accessible date/time pickers + slot generator core

### PR DESCRIPTION
**Pull request by lukevella**
## What this is

The **flow-agnostic foundation** split out of the slot-generator work. During that session we realised the generator's *flow* (a modal with weekday recurrence) isn't the optimal poll-creation UX — the poll-creation flow is going to be redesigned (calendar-first, canvas + accessible sidebar). Rather than ship a UI we're about to replace, this PR lands only the **reusable pieces** so they don't rot on a branch, while the flow-specific shell stays parked.

## What's included

**`packages/ui` — accessible segmented pickers (reusable anywhere):**
- `DateRangePicker` — react-aria segmented date range field: each part is its own spinbutton (arrow keys, typeahead), fully styleable, no native picker. Localized via react-aria `I18nProvider`; `firstDayOfWeek` respected. Popover range calendar with a single/multi-month view.
- `TimeField` — segmented time field, same model; honours `hourCycle`.
- `date-field-styles.ts` — shared field/segment styling (matches Input geometry).
- `input-group.tsx` — height fix (`h-8` → `h-9`) to match the standard Button/Input.

**`lib/datetime`:**
- `slot-generator.ts` — the isomorphic slot generator, moved out of the private API route so it's importable app-wide (same code powers the REST `/api/private` poll create).
- Short weekday names added to `locales.ts` via `Intl` (localized, no hardcoded strings).

**REST poll route:**
- `MAX_POLL_OPTIONS` single-sourced into a shared constant.

## What's NOT here (parked)

The generator dialog, weekday-recurrence toggles, and their form wiring stay on `claude/slot-generation-modal-8ba2b4`. Nothing in the current poll-creation flow is exposed by this PR — the new UI components are unused entry points until the redesigned flow consumes them.

## Notes

- Verified: no parked file leaked; the landed pickers have no consumer on this branch by design (UI-lib entry points).
- 304 unit tests pass, type-check clean in both packages, Biome clean (one pre-existing unrelated warning).
- **Not visually verified** — the in-app browser tool was unavailable during the session, so the pickers' rendering (segment alignment, selected-range styling, localized calendars) was validated by reading computed classes and the react-aria contract, not by eye. Worth a manual pass before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added an app-wide locale provider so date/time components use the selected locale.
  - Introduced accessible segmented date-range picker and segmented time field, with shared styling for segments and triggers.
  - Added abbreviated weekday names alongside full labels.
- **Bug Fixes**
  - Enforced a shared maximum of 100 poll options and updated “too many options” messaging.
  - Added guardrails for time-slot generation: capped oversized date ranges and rejected invalid duration/interval inputs (with clearer 400 errors).
- **Style**
  - Standardized input group height and refined date/time field styling for focus/invalid/disabled states.

### CODE REVIEW

File: apps/web/src/lib/datetime/slot-generator.ts [hunk 1]
```diff
@@ -72,12 +76,17 @@ export const generateTimeSlots = (
...
+  let daysVisited = 0;
 
   for (
     let cursor = startDay;
     cursor.isSameOrBefore(endDay, "day");
     cursor = cursor.add(1, "day")
   ) {
+    if (daysVisited >= MAX_SLOT_GENERATION_DAYS) {
+      break;
+    }
```
[the remark this answers is not in the document]
Comment (lukevella) [reply apps/web/src/lib/datetime/slot-generator.ts:88]: Fixed in 40c753f2: `slotGeneratorSchema` now rejects any range spanning ≥ `MAX_SLOT_GENERATION_DAYS` (or `endDate` < `startDate`) via a `.refine`, so the `validator("json", …)` middleware returns a 400 before generation runs — no silent truncation. The bound matches the generator exactly (a span the generator can fully produce passes; one it would truncate is rejected). Regression tests added in 0d6dacc6 (route: over-long/reversed/boundary cases assert 400 + error targets `endDate`; generator: caps iteration at the limit).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

[the remark this answers is not in the document]
Comment (lukevella) [reply apps/web/src/lib/datetime/slot-generator.ts:88]: Thanks for confirming — no further change needed. The schema `.refine` (reject reversed ranges and spans ≥ `MAX_SLOT_GENERATION_DAYS` before generation) plus the route/generator regression tests are already in place, so this thread is complete.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
